### PR TITLE
feat: event team Assignment tab atom

### PR DESCRIPTION
**Pull request by ibex088**
## What does this PR do?

Screenshot 2024-10-17 at 12 55 06 PM

- Fixes #XXXX (GitHub issue number)
- Fixes CAL-XXXX (Linear issue number - should be visible at the bottom of the GitHub issue description)

## Mandatory Tasks (DO NOT REMOVE)

- [ ] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

### CODE REVIEW

File: apps/api/v2/src/modules/atoms/services/event-types-atom.service.ts [hunk 1]
```diff
@@ -45,12 +46,51 @@ export class EventTypesAtomService {
       throw new NotFoundException(`Event type with id ${eventTypeId} not found`);
     }
 
-    this.eventTypeService.checkUserOwnsEventType(user.id, eventType.eventType);
+    if (eventType?.team?.id) {
+      await this.checkTeamOwnsEventType(user.id, eventType.eventType.id, eventType.team.id);
```
Comment (ThyMinimalDev) [inline apps/api/v2/src/modules/atoms/services/event-types-atom.service.ts:50]: we also need to make sure the user is team owner or admin

File: apps/api/v2/src/modules/atoms/services/event-types-atom.service.ts [hunk 2]
```diff
@@ -77,4 +117,30 @@ export class EventTypesAtomService {
 
     return eventType.eventType;
   }
+
+  async checkCanUpdateTeamEventType(userId: number, eventTypeId: number, teamId: number, scheduleId: number) {
+    await this.checkTeamOwnsEventType(userId, eventTypeId, teamId);
+    await this.teamEventTypeService.validateEventTypeExists(teamId, eventTypeId);
+    await this.eventTypeService.checkUserOwnsSchedule(userId, scheduleId);
+  }
```
[the remark this answers is not in the document]
Comment (SomayChauhan) [reply apps/api/v2/src/modules/atoms/services/event-types-atom.service.ts:125]: https://github.com/calcom/cal.com/pull/17145/commits/78ea9f19b214850142dce1aa0b89c31ec9e46054

File: apps/api/v2/src/modules/atoms/services/event-types-atom.service.ts [hunk 3]
```diff
@@ -77,4 +117,31 @@ export class EventTypesAtomService {
...
+      select: {
+        team: {
+          select: {
+            eventTypes: true,
+          },
+        },
+      },
+    });
+    if (!membership?.team?.eventTypes?.some((item) => item.id === eventTypeId)) {
+      throw new ForbiddenException(`Team with ID=${teamId} does not own event type with ID=${eventTypeId}`);
```
Comment (ThyMinimalDev) [inline apps/api/v2/src/modules/atoms/services/event-types-atom.service.ts:144]: update error message to reflect that users with role MEMBER are not allowed to access the endpoint

Comment (SomayChauhan) [reply apps/api/v2/src/modules/atoms/services/event-types-atom.service.ts]: https://github.com/calcom/cal.com/pull/17145/commits/22aaa94efb88ccdf3d41624ca371ec759f04cbe2

**Review by ThyMinimalDev — APPROVED** [review]

**Pull request merged**